cluster/addons/addon-manager/kube-addon-update.sh

```diff
@@ -262,7 +260,11 @@ function create-object() {
     log INFO "Creating new ${obj_type} from file ${file_path} in namespace ${namespace}, name: ${obj_name}"
     # this will keep on failing if the ${file_path} disappeared in the meantime.
     # Do not use too many retries.
-    run-until-success "${KUBECTL} create --namespace=${namespace} -f ${file_path}" ${NUM_TRIES} ${DELAY_AFTER_ERROR_SEC}
+    if [[ -n "${namespace}" ]]; then
+        run-until-success "${KUBECTL} create --namespace=${namespace} -f ${file_path}" ${NUM_TRIES} ${DELAY_AFTER_ERROR_SEC}
```

Comment:
If you create an object with out --namespace and it declares a non default namespace in the yml, does it use the namespace from the yml? If so (which I thought was the case) you could just remove the namespace arg